tests/testmacro.nim

```diff
@@ -136,6 +136,93 @@ suite "Macro transformations test suite":
     check:
       waitFor(gen(int)) == default(int)
 
+  test "Nested return":
+    proc nr: Future[int] {.async.} =
+      return
+        if 1 == 1:
+          return 42
+        else:
+          33
+
+    check waitFor(nr()) == 42
+
+  test "Issue #415 (run closure to completion on return)":
+    var x = 0
```

Comment:
let's use a fresh variable for each test, this ends up being a global which is annoying for gcsafety down the line

Reply:
nevermind, missed that it's in a `test`

Reply:
Yeah, could make it a suite instead if you want (and change `x` into a var)

Reply:
that's probably more clean, yes